keps/sig-storage/1682-csi-driver-skip-permission/README.md

```diff
@@ -0,0 +1,238 @@
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [x] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [x] (R) KEP approvers have approved the KEP status as `implementable`
+- [x] (R) Design details are appropriately documented
+- [x] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [x] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+
+## Summary
+
+Currently before bind mounting a CSI volume for a Pod, we use imprecise heuristics,
+such as presence of fsType on the PVC to determine if the volume supports fsGroup based
+permission change. These heuristics are known to be fragile, and cause problems with different
+storage types.
+
+To solve this issue we will add a new field called `CSIDriver.Spec.SupportsFSGroup` 
+that allows the driver to define if it supports volume ownership modifications via
+fsGroup.
+
+## Motivation
+
+When a volume is mounted on the node, we modify volume ownership 
+before bind mounting the volume inside container; however, if the volume
+does not support these operations (such as NFS), then the change is still attempted. 
+This results in errors being reported to the user if the volume doesn't 
+support these operations.
+
+### Goals
+
+ - Allow CSIDrivers to opt-in to volume ownership changes.
+
+### Non-Goals
+
+ - This does not attempt to update existing CSIDrivers themselves.
+
+## Proposal
+
+We propose that the `CSIDriver` type include a field that defines if the volume 
+provided by the driver supports changing volume ownership. This will be enabled
+with a new feature gate, `CSIVolumeSupportFSGroup`.   
+
+### Risks and Mitigations
+
+- The CSIDriver objects will need to be redeployed after this field is introduced if the desired behavior is modified.
+- If a cluster enables the `CSIVolumeSupportFSGroup` feature gate and then this feature gate is disabled,
+such as due to an upgrade or downgrade, then the cluster will revert to the current behavior of examining
+volumes and attempting to apply volume ownerships and permissions based on the defined `fsGroup`.
+
+## Design Details
+
+Currently Volume permission and ownership change is done automatically for every volume. 
```

Comment:
nit: s/Volume/volume

Reply:
Also yes this line is not right. We don't change permission and ownership of every volume by default.

Reply:
Please update this.

Reply:
This has been updated. The behavior described in the API comments is correct - we only change permissions and ownership if `fsGroup` is defined and accessModes is RWO.